### PR TITLE
Issue 11 reauthentication POC

### PR DESCRIPTION
**Pull request by rouazana**
Here is a reauthentication POC, used on the simpler method: GetMailboxes.
Please check if it's enough fine to be generalized.

### CODE REVIEW

File: src/main/scala-2.11/org/apache/james/gatling/jmap/JmapMailboxes.scala [hunk 1]
```diff
@@ -3,30 +3,27 @@ package org.apache.james.gatling.jmap
...
-  def getSystemMailboxes = {
-    val inboxIdPath = s"$$$mailboxListPath[?(@.role == 'inbox')].id"
-    val outboxIdPath = s"$$$mailboxListPath[?(@.role == 'outbox')].id"
-    val sentIdPath = s"$$$mailboxListPath[?(@.role == 'sent')].id"
-    getMailboxes
-      .check(jsonPath(inboxIdPath).saveAs("inboxMailboxId"))
-      .check(jsonPath(outboxIdPath).saveAs("outboxMailboxId"))
-      .check(jsonPath(sentIdPath).saveAs("sentMailboxId"))
-  }
+  def getSystemMailboxes = getMailboxes
```
Comment (chibenwa) [inline src/main/scala-2.11/org/apache/james/gatling/jmap/JmapMailboxes.scala:23]: Does line 23 imples all mailbox is system mailbox ?

**Review by chibenwa — COMMENTED** [review]
Overall, I agree with this POC
Just, in my opinion, to be a complete :+1:, I would remove now useless getSystemMailbox method....

File: src/main/scala-2.11/org/apache/james/gatling/jmap/JmapMailboxes.scala (hunk 1 above)
Comment (rouazana) [reply src/main/scala-2.11/org/apache/james/gatling/jmap/JmapMailboxes.scala:23]: Of course not. It just that it uses the same request.
I wanted to keep one request / one checks set for each available request, but I can remove it if you find it better

**Review by chibenwa — APPROVED** [review]

**Review by mbaechler — APPROVED** [review]

**Review by aduprat — APPROVED** [review]

File: src/main/scala-2.11/org/apache/james/gatling/utils/RetryAuthentication.scala [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+  def execWithRetryAuthentication(scenario: HttpRequestBuilder, checks: Seq[HttpCheck]): ChainBuilder =
+    exec(scenario.check(status.in(200, 401).saveAs("statusCode")).check(checkIfOk(checks): _*))
+      .doIfEquals("${statusCode}", 401){
+          JmapAuthentication.authentication()
+            .exec(scenario.check(checks: _*))
+      }
+
+  def checkIfOk(checks: Seq[HttpCheck]): Seq[HttpCheck] =
+    checks.map(check => checkIf((r: Response, s: Session) => r.statusCode == 200){check})
```
Comment (chibenwa) [inline src/main/scala-2.11/org/apache/james/gatling/utils/RetryAuthentication.scala:22]: r.statusCode is an Option, and not an int. Hence, it is always false and test were never performed....

**Comment by chibenwa** [comment]
Move here : https://github.com/linagora/james-gatling/pull/29

**Pull request closed without merging**